### ISSUE TITLE
Erroneous file descriptions and a push related problem.

### ISSUE THREAD

**Issue by S3NTYN3L** [Cube777/dotgit#4]
Have a look at ~~[my test repo]~~ (link removed).
As you can see, I've three hosts; `GLaDOS`, `REDQUEEN` and `SHODAN`.

After running the `dotgit update` command and upon issuing the `dotgit generate` command, the output is;
```
Changes: added GLadeletedOS/.bashrc; added renamedEdeletedqueen/.bashrc; added SHOdeletedaddedN/.bashrc
...etc...
```

Now, if I set the hostname to use all lowercase characters, and edit the `filelist` accordingly, the erroneous output doesn't appear.
Can you please correct this issue for those of us that use uppercase characters?

Secondly, if I type the wrong username or password when prompted by `dotgit generate`, the push obviously fails.
However, issuing the command a second time outputs;
```
Running safeguards...
No changes to repository...
```
and exits.

My only recourse is to make an additional change to any files to be pushed, run `dotgit update`, undo my additional changes, run `dotgit update` yet again and finally run `dotgit generate` again with, hopefully, the correct credentials.
I'm sure you can see the problem with that.

Thanks in advance! ;)

**Comment by Cube777** [comment]
Hi @S3NTYN3L, thanks for using dotgit :) The first problem you described is definitely wrong - I'm planning to rewrite dotgit in the coming weeks to be more robust and this should fix the first issue :) As for the second issue this is because dotgit checks for uncommitted changes, not unpushed changes. Since the dotgit repo is a normal git repo you can then issue the command 'git push' to re-initiate the repo pushing and it should work perfectly (without any additional changes) but I can see the problem with this and will address this in the new version. Thanks for reporting the bugs!

**Comment by S3NTYN3L** [comment]
Sweet!
I look forward to seeing what you've come up with! ;)

**Comment by Cube777** [comment]
Hey @S3NTYN3L   , I finally got around to rewriting dotgit and both the issues that you mentioned (erroneous commit messages and not pushing even though there is unpushed commits) have been addressed. Please have a look at [this post](https://bbs.archlinux.org/viewtopic.php?pid=1720722#p1720722) for instructions. Have a good day!

**Comment by Cube777** [comment]
Fixed in 1.4.1
